chrome/android/java/src/org/chromium/chrome/browser/preferences/Preferences.java

```diff
@@ -189,6 +196,10 @@ protected void onCreate(Bundle savedInstanceState) {
     protected void onDestroy() {
         super.onDestroy();
 
+        if (AdblockMode.sManageSubscriptions) {
```

Comment:
> What about checking whether subscriptionsManager is null? It can be the second condition in the if.

It can't be `null`, it's created when in `onCreate()` the condition is checked. There is no other places to release it before. `onDestroy` happens only once

Reply:
If we just check the variable if it's `null` and do nothing it can hide another issue - it's not even created in `onCreate`. So as for me we should check for the original reason (subscriptions management is enabled), not the consequence (the instance is created).

Reply:
We can have both, just a safety check before dereferencing it.

Reply:
in current logics `instance != null` is absolutely equal and only equal to condition `AdblockMode.sManageSubscriptions == true`. I don't think we should overengineer it or broaden to any other use cases.

Reply:
Then I would leave it to the android team.